src/meta/src/hummock/compaction_group/manager.rs

```diff
@@ -282,14 +294,34 @@ impl CompactionGroupManagerInner {
 
     async fn register<S: MetaStore>(
         &mut self,
-        pairs: &[(StateTableId, CompactionGroupId, TableOption)],
+        pairs: &mut [(StateTableId, CompactionGroupId, TableOption)],
         meta_store: &S,
     ) -> Result<Vec<StateTableId>> {
+        let mut current_max_id = StaticCompactionGroupId::END as CompactionGroupId;
+        if let Some((&k, _)) = self.compaction_groups.last_key_value() {
+            if k > current_max_id {
+                current_max_id = k;
+            }
+        };
         let mut compaction_groups = BTreeMapTransaction::new(&mut self.compaction_groups);
-        for (table_id, compaction_group_id, table_option) in pairs {
-            let mut compaction_group = compaction_groups
-                .get_mut(*compaction_group_id)
-                .ok_or(Error::InvalidCompactionGroup(*compaction_group_id))?;
+        for (table_id, compaction_group_id, table_option) in pairs.iter_mut() {
+            let mut compaction_group =
+                if *compaction_group_id == StaticCompactionGroupId::NewCompactionGroup as u64 {
+                    current_max_id += 1;
+                    *compaction_group_id = current_max_id;
+                    compaction_groups.insert(
```

Comment:
Need initialize a CompactStatus for this new group too.
Previously we initialize all predefined groups on system start https://github.com/risingwavelabs/risingwave/blob/91f53d8fbf614b79eadeace1499e94e7a33e4efa/src/meta/src/hummock/manager/mod.rs#L270, but now we may initialize lazily.

Reply:
> Need initialize a CompactStatus for this new group too. Previously we initialize all predefined groups on system start
> 
> https://github.com/risingwavelabs/risingwave/blob/91f53d8fbf614b79eadeace1499e94e7a33e4efa/src/meta/src/hummock/manager/mod.rs#L270
> 
> , but now we may initialize lazily.

fixed